src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -23,22 +26,32 @@
 ;; ## Plugin Initialization
 
 (defn initialize!
-  "Initialize the plugin state."
+  ;; should really just be named initialize, not initialize!, all mutation is self contained
+  ; 1. Access app info using app accessor to determine url, app-id, etc.
+  ; 2. Create a vault client
+  ; 3. Authenticate the vault client
+  ; 4. Return the vault client
+  "Set up the vault client."
   [logger app-accessor]
   (alter-var-root #'log/logger (constantly logger))
-  ;; TODO: determine what goes in the state
+  ;; TODO: actually initialize a Vault client
   (atom {}))
```

Comment:
Since you've removed atom references elsewhere, maybe return `nil` for now instead?